components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -1812,125 +1816,334 @@ private int handleStartIndexEqualsNULL(Integer startIndex) {
         return startIndex;
     }
 
-    private List<Object> listGroups(Map<String, Boolean> requiredAttributes) throws CharonException {
+    /**
+     * List all the groups.
+     *
+     * @param startIndex         Start index in the request.
+     * @param count              Limit in the request.
+     * @param sortBy             SortBy
+     * @param sortOrder          Sorting order
+     * @param domainName         Domain Name
+     * @param requiredAttributes Required attributes
+     * @return
+     * @throws CharonException
+     */
+    private List<Object> listGroups(int startIndex, Integer count, String sortBy, String sortOrder, String domainName,
+            Map<String, Boolean> requiredAttributes) throws CharonException {
+
         List<Object> groupList = new ArrayList<>();
         //0th index is to store total number of results;
         groupList.add(0);
         try {
-            SCIMGroupHandler groupHandler = new SCIMGroupHandler(carbonUM.getTenantId());
-            Set<String> roleNames = groupHandler.listSCIMRoles();
+            Set<String> roleNames = getRoleNamesForGroupsEndpoint(domainName);
```

Comment:
Can we have this method as getRoleNames

Reply:
We already have a method as getRoleNames() and it is used to get roles for both users groups endpoint.

Reply:
Ack